Packs/1Password/Integrations/OnePasswordEventCollector/OnePasswordEventCollector_test.py

```diff
@@ -0,0 +1,323 @@
+    """
+    Given:
+        - A OnePasswordEventCollector.Client instance with valid inputs to the get_events method.
+
+    When:
+        - Calling Client.get_events.
+
+    Assert:
+        - Ensure no exception is raised and the raw API response is as expected.
+    """
+    from OnePasswordEventCollector import urljoin
+
+    event_type = 'Sign in attempts'
+    from_date = datetime(2024, 12, 2, 11, 50)
+
+    event_url = urljoin(BASE_URL, '/signinattempts')
+    mock_response = util_load_json('test_data/signinattempts_response.json')
+    requests_mock.post(event_url, json=mock_response)
+
+    with assert_not_raises(Exception):
+        response = authenticated_client.get_events(event_type=event_type, from_date=from_date)
+
+    assert response == mock_response
+
+
+@pytest.mark.parametrize(
+    'event_type, error_message',
+    [
+        pytest.param(
+            'Random event',
+            'Invalid or unsupported 1Password event type: Random event.',
+            id='Invalid Event Type',
+        ),
+        pytest.param(
+            'audit events',
+            "Either a 'pagination_cursor' or a 'from_date' need to be specified.",
+            id='Valid event type but missing other params',
+        )
+    ]
+)
+def test_client_get_events_invalid_inputs(authenticated_client: Client, event_type: str, error_message: str):
+    """
+    Given:
+        - Case 1: A OnePasswordEventCollector.Client instance with an invalid event type.
+        - Case 2: A OnePasswordEventCollector.Client instance with an valid event type (but missing other params).
+
+    When:
+        - Calling Client.get_events.
+
+    Assert:
+        - Ensure a ValueError is raised with the appropriate error message.
+    """
+    with pytest.raises(ValueError, match=error_message):
+        authenticated_client.get_events(event_type)
+
+
+def test_get_events_from_client(authenticated_client: Client, mocker: MockerFixture):
+    """
+    Given:
+        - A 1Password event type, from date, and the maximum number of events.
+
+    When:
+        - Calling get_events_from_client (which calls Client.get_events).
+
+    Assert:
+        - Ensure Client.get_events is called twice (because first response['has_more'] is True).
+        - Ensure the number of events does not exceed the specified maximum and the events are as expected.
+    """
+    from OnePasswordEventCollector import get_events_from_client
+
+    event_type = 'audit events'
+    from_date = datetime(2024, 12, 2, 11, 50)
+    max_events = 20
+
+    client_get_events = mocker.patch.object(authenticated_client, 'get_events', side_effect=mock_client_get_events)
+
+    events = get_events_from_client(authenticated_client, event_type=event_type, from_date=from_date, max_events=max_events)
+
+    expected_events = util_load_json('test_data/auditevents_expected_events.json')
+
+    assert client_get_events.call_count == 2
+    assert len(events) <= max_events  # Sanity check
```

Comment:
better do a `==` with the expected value than `<=`, since the value here is constant